tests/infrastructure/test_data_manager_stack.py

```diff
@@ -26,3 +32,824 @@ def test_sds_data_manager_stack(app, sds_id):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "FunctionName": f"file-indexer-{sds_id}",
+            "Runtime": "python3.9",
+            "Handler": "SDSCode.indexer.lambda_handler",
+            "MemorySize": 1000,
+            "Timeout": 15 * 60,
+        },
+    )
+
+
+def test_upload_api_lambda_function_resource_properties(template, sds_id):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "FunctionName": f"upload-api-handler-{sds_id}",
+            "Runtime": "python3.9",
+            "Handler": "SDSCode.upload_api.lambda_handler",
+            "MemorySize": 1000,
+            "Timeout": 15 * 60,
+        },
+    )
+
+
+def test_query_api_lambda_function_resource_properties(template, sds_id):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "FunctionName": f"query-api-handler-{sds_id}",
+            "Runtime": "python3.9",
+            "Handler": "SDSCode.queries.lambda_handler",
+            "MemorySize": 1000,
+            "Timeout": 60,
+        },
+    )
+
+
+def test_download_api_lambda_function_resource_properties(template, sds_id):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "FunctionName": f"download-query-api-{sds_id}",
+            "Runtime": "python3.9",
+            "Handler": "SDSCode.download_query_api.lambda_handler",
+            "Timeout": 60,
+        },
+    )
+
+
+def test_aws_lambda_function_resource_properties(template):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "Handler": "index.handler",
+            "Runtime": "nodejs14.x",
+            "Timeout": 120,
+            "Role": {
+                "Fn::GetAtt": [
+                    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
+                    "Arn",
+                ]
+            },
+        },
+    )
+
+
+def test_aws_bucket_notification_lambda_function_resource_properties(template):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "Handler": "index.handler",
+            "Runtime": "python3.9",
+            "Timeout": 300,
+            "Role": {
+                "Fn::GetAtt": [
+                    "BucketNotificationsHandler050a0587b7544547bf325f094a3db834RoleB6FB88EC",
+                    "Arn",
+                ]
+            },
+        },
+    )
+
+
+def test_custom_s3_auto_delete_lambda_function_resource_properties(template):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "Handler": "__entrypoint__.handler",
+            "Runtime": "nodejs14.x",
+            "Timeout": 900,
+            "MemorySize": 128,
+            "Role": {
+                "Fn::GetAtt": [
+                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
```

Comment:
Same here, do we want this specific role checked or is it an auto-generated hash we aren't too concerned with?

Reply:
Maybe I'll change this to Match.string_like_regexp(CustomS3AutoDeleteObjectsCustomResourceProviderRole*)? It might be good to check that there's a role that's being created for the S3 deletion lambda? But that role is auto-generated.

Reply:
That's a good idea. I'll use string_like_regexp() for the other one